docker/buildDocker.sh

```diff
@@ -43,135 +64,102 @@ parseCommandLineArgs()
 	echo "			--jdk-boot-dir|-J			Specify the boot JDK directory"
 	echo "			--openj9|-j9				Builds using OpenJ9 instead of Hotspot"
 	echo "			--use-eclipse-docker-files|-e		Builds the specified jdk using the Eclipse Openj9 dockerfiles"
 	echo
 }
 
-checkJDKVersion()
-{
-	case "$jdkVersion" in
-		"jdk8u" | "jdk8" | "8" | "8u" )
-			jdkVersion="jdk8u";;
-		"jdk9u" | "jdk9" | "9" | "9u" )
-			jdkVersion="jdk9u";;
-		"jdk10u" | "jdk10" | "10" | "10u" )
-			jdkVersion="jdk10u";;
-		"jdk11u" | "jdk11" | "11" | "11u" )
-			jdkVersion="jdk11u";;
-		"jdk12u" | "jdk12" | "12" | "12u" )
-			jdkVersion="jdk12u";;
-		"jdk13u" | "jdk13" | "13" | "13u" )
-			jdkVersion="jdk13u";;
-                "jdk14u" | "jdk14" | "14" | "14u" )
-                        jdkVersion="jdk14";;
-                "jdk" | "jdknext" )
-                        jdkVersion="jdk";;
-		"all" )
-			jdkVersion="jdk8u jdk9u jdk10u jdk11u jdk12u jdk13u jdk14u jdk";;
-		*)
-			echo "Not a valid JDK Version" ; jdkVersionList; exit 1;;
-	esac
-}
-
-jdkVersionList()
-{
-	echo
-	echo "Valid JDK versions :
-		- jdk8u
-		- jdk9u
-		- jdk10u
-		- jdk11u
-		- jdk12u
-		- jdk13u
-		- jdk14u
-		- jdk"
+checkJDK() {
+  if ! ((JDK_VERSION >= 8 && JDK_VERSION <= JDK_MAX)); then
+    echo "Please input a JDK between 8 & ${JDK_MAX}, or 'jdk'"
+    exit 1
+  fi
 }
 
 checkArgs()
 {
-	# Sets WORKSPACE to home if WORKSPACE is empty or unbounded
-	if [ ! -n "${WORKSPACE:-}" ]; then
-        	echo "WORKSPACE not found, setting it as environment variable 'HOME'"
-        	WORKSPACE=$HOME
+	# ${WORKSPACE##*/} returns the name of the current dir
+	if [ ${WORKSPACE##*/} != "docker" ]; then
+		echo "Unable to run script from : $WORKSPACE"
+		echo "The script must be run from openjdk-build/docker/"
+		exit 1
 	fi
-	if [ "$cleanWorkspace" == true ]; then
-		echo "Cleaning workspace"
-		rm -rf $WORKSPACE/DockerBuildFolder/$jdkVersion-$buildVariant
+	if [ "$CLEAN" == true ]; then
+		echo "Removing all jdkXX folders"
+		rm -rf $WORKSPACE/jdk*
+		echo "Removing old workspace folder"
+		rm -rf $WORKSPACE/workspace
+		echo "Removing old EclipseDockerfiles folder"
+		rm -rf $WORKSPACE/EclipseDockerfiles
 	fi
 }
 
 useEclipseOpenJ9DockerFiles()
 {
-	mkdir -p $WORKSPACE/DockerBuildFolder/$jdkVersion-$buildVariant/EclipseDockerfiles
-	cd $WORKSPACE/DockerBuildFolder/$jdkVersion-$buildVariant/EclipseDockerfiles
+	local dockerfileDir="$WORKSPACE/EclipseDockerfiles"
+	local jdk="jdk"
+
+	mkdir -p $dockerfileDir
+	cd $dockerfileDir
 	wget https://raw.githubusercontent.com/eclipse/openj9/master/buildenv/docker/mkdocker.sh
 	chmod +x mkdocker.sh
 	# Generate an Ubuntu1804 Dockerfile using mkdocker.sh
-	$WORKSPACE/DockerBuildFolder/$jdkVersion-$buildVariant/EclipseDockerfiles/mkdocker.sh --dist=ubuntu --version=18 --print >> $WORKSPACE/DockerBuildFolder/$jdkVersion-$buildVariant/EclipseDockerfiles/Dockerfile
+	$dockerfileDir/mkdocker.sh --dist=ubuntu --version=18 --print >> $dockerfileDir/Dockerfile
```

Comment:
do we want to pass in 18 as a parameter?

Reply:
https://github.com/AdoptOpenJDK/openjdk-build/pull/1911#discussion_r444010951 
For now, it's hardcoded to stay `18` as all of the Dockerfiles generated using `dockerfile-generator.sh` can only be Ubuntu1804, so I think the `mkDocker.sh` generated dockerfiles should be consistent with those :-)

Reply:
@karianna Are you ok with that explanation? Would be good to merge this and test the merge overnight if possible

Reply:
I'll live with it :-)